camel/tasks/task.py

```diff
@@ -0,0 +1,257 @@
+from camel.messages import BaseMessage
+from camel.prompts import TextPrompt
+
+from .task_prompt import TASK_DECOMPOSE_PROMPT, TASK_EVOLVE_PROMPT
+
+TASK_STATES = {"OPEN", "RUNNING", "DONE", "DELETED"}
+
+
+class Task(BaseModel):
+    """
+    Task is specific assignment that can be passed to a agent.
+
+    Attributes:
+        content: string content for task.
+        id: A unique id.
+        state: The state which should be OPEN, RUNNING, DONE or DELETED.
+        parent: The parent task, None for root task.
+        subtasks: The childrent sub-tasks for the task.
+        result: The answer for the task.
+    """
+
+    content: str
+    """The content of the task."""
+
+    id: str
+    """An unique string identifier for the task. This should ideally be
+    provided by the provider/model which created the task."""
+
+    state: Literal["OPEN", "RUNNING", "DONE", "DELETED"] = "OPEN"
+    """The task state.
+    """
+
+    parent: Optional["Task"] = None
+    """The parent task.
+    """
+
+    subtasks: List["Task"] = []
+    """A list of sub tasks.
+    """
+
+    result: str = ""
+
+    @classmethod
+    def from_message(cls, message: BaseMessage) -> "Task":
+        r"""Create a task from a message.
+
+        Args:
+            message (BaseMessage): The message to the task.
+
+        Returns:
+            Task
+        """
+        return cls(content=message.content, id="0")
+
+    @staticmethod
+    def to_message():
+        r"""Convert a Task to a Message."""
+        # TODO
+        pass
+
+    def reset(self):
+        r"""Reset Task to initial state."""
+        self.state = "OPEN"
+        self.result = ""
+
+    def update_task_result(self, message: BaseMessage):
+        self.result = message.content
+        self.state = "DONE"
+
+    def set_id(self, id: str):
+        self.id = id
+
+    def set_state(self, state: Literal["OPEN", "RUNNING", "DONE", "DELETED"]):
+        assert state in TASK_STATES, f"Invalid state `{state}`"
+
+        self.state = state
+        if state == "DONE":
+            for subtask in self.subtasks:
+                if subtask.state != "DELETED":
+                    subtask.set_state(state)
+        elif state == "RUNNING":
+            if self.parent is not None:
+                self.parent.set_state(state)
+
+    def add_subtask(self, task: "Task"):
+        task.parent = self
+        self.subtasks.append(task)
+
+    def get_running_task(self) -> Optional["Task"]:
+        for sub in self.subtasks:
+            if sub.state == "RUNNING":
+                return sub.get_running_task()
+        if self.state == "RUNNING":
+            return self
+        return None
+
+    def to_string(self, indent="") -> str:
+        _str = f"{indent}Task {self.id}: {self.content}\n"
+        for subtask in self.subtasks:
+            _str += subtask.to_string(indent + "  ")
+        return _str
+
+
+def parse_response(response: str, task_id: Optional[str]) -> List[Task]:
```

Comment:
It looks like that the relationship between parent task and sub tasks is not considered when we generate sub tasks.

Reply:
For `id=f"{task_id}.{i}"`, `{task_id}` is the id of parent task and `.{i}` means the id of sub tasks.